kuiper-game/Domain/Ship.cs

```diff
@@ -1,50 +1,85 @@
 using System;
+using Kuiper.Domain.CelestialBodies;
 
 namespace Kuiper.Domain
 {
     public class Ship
     {
-        public Ship(string name, string shipClass, int speed)
+        private const double EARTH_GRAVITY = 9.80665;
```

Comment:
it should be a const somewhere else thats true. Its not used to calculate acceleration though, that would be wetmass over thrust. AccelerationGs does take it into account since its relative to what humans (from earth) are comfortable with

Reply:
I misunderstood. 
I would still consider taking an iengine in the ctor and whatever shipclass even (its a string now but perhaps could be some concrete static objects to "choose" from that had different weight classes etc?, and letting something else do all the calculations. 
So the domain objects just hold the information but dont do any calculations (Like accelaration).

Reply:
What is gained by making the domain objects more anemic? Acceleration is "business logic" for a Ship, taking into consideration it's modules, fuel, own weight (adding up total weight) and it's engines capabilities. Since all these exists as part of the ship, they can give the Ship class all the info it needs to be able to calculate it's current acceleration capabilities.

Looking at Evans and Fowlers input here: https://martinfowler.com/eaaCatalog/domainModel.html i interpret this as it being a good idea to use relationships between Modules, Engine and Ship and then do the calculations on the Ship class. The acceleration calculation MIGHT make sense elsewhere, but it's much easier to understand by having it clearly defined how the SHIP calculates it. 

The alternative could be a generic physics library, i'm not sure i think that is preferable here, we aren't quite in "Math.*" lib territory. 

Currently a Ship takes a ShipEngine in it's constructor (not an IShipEngine though) and class is a remnant to be changed to something else when that concept is ready: https://github.com/mifriis/edge-of-kuiper/pull/39/files#diff-35cc78be9f73d67a92ce2b4c5e0db19e174984f14077259febea8bf2d22eb76fR9 is the IShipEngine to make testing easier by being able to mock it's output too?

Reply:
It was merely to enable ships to be created with different types of engines, paving the way for upgrades. 

It all depends on how you see the domain i guess. 
You are the domain expert in this case, so if you think it makes more sense having it inside, then keep it inside. 

Personally i dont think its the ships problem, specifically to calculate its trajectory, speed, where it is, etc. I would try to encapsulate that somewhere. Where i am not sure though.

Reply:
> Personally i dont think its the ships problem, specifically to calculate its trajectory, speed, where it is, etc. I would try to encapsulate that somewhere. Where i am not sure though.

It doesn't do that though. Ship provides `getters` for it's capabilities. 
* It's maximum acceleration based on it's current mass and engine, _not_ it's current travelling acceleration. Similar to your car's 0-100km/h metric or top speed. 
* DeltaV is the calculation of "how many kms can the car travel with it's current tank" that shows up in your dashboard when driving.

If i were to model a Car in DDD i would put these things on the Car.cs, perhaps using an Engine.cs or a FuelTank.cs as input to the calculations. Where it's headed and things like that belong in the "GPS-Service" or similiar.

However things like where the ship is, if it can go there, if it should start the journey, how fast is it currently travelling along that route and if it's arrived there yet is all handled by the ShipService which sounds a bit similiar to the Navigation API you mention. See `interface IShipService`